### PR TITLE
Start fixing lint errors, golangci-lint had bunch of linters disabled

### PR DESCRIPTION
**Pull request by bogdandrutu**
Updates #2789 

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

**Review by tigrannajaryan — COMMENTED** [review]
LGTM, except I am not sure I like that it forces every exported symbol to have a comment. What do you think?

**Comment by bogdandrutu** [comment]
> LGTM, except I am not sure I like that it forces every exported symbol to have a comment. What do you think?

I think we should do that, we can exclude `internal` packages but in general all our public API should be documented.

**Review by Aneurysm9 — APPROVED** [review]

**Review by carlosalberto — APPROVED** [review]

**Comment by bogdandrutu** [comment]
@tigrannajaryan ping on this :) I need a codeowner approval

**Review by tigrannajaryan — APPROVED** [review]

**Pull request merged**
